builtin/logical/pki/ocsp.go

```diff
@@ -0,0 +1,397 @@
+package pki
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/helper/errutil"
+
+	"golang.org/x/crypto/ocsp"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/certutil"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	ocspReqParam            = "req"
+	ocspResponseContentType = "application/ocsp-response"
+)
+
+type ocspRespInfo struct {
+	formattedSerialNumber string
+	serialNumber          *big.Int
+	ocspStatus            int
+	revocationTimeUTC     *time.Time
+	issuerID              issuerID
+}
+
+// These response variables should not be mutated, instead treat them as constants
+var (
+	OcspUnauthorizedResponse = &logical.Response{
+		Data: map[string]interface{}{
+			logical.HTTPContentType: "application/ocsp-response",
+			logical.HTTPStatusCode:  http.StatusUnauthorized,
+			logical.HTTPRawBody:     ocsp.UnauthorizedErrorResponse,
+		},
+	}
+	OcspMalformedResponse = &logical.Response{
+		Data: map[string]interface{}{
+			logical.HTTPContentType: "application/ocsp-response",
+			logical.HTTPStatusCode:  http.StatusBadRequest,
+			logical.HTTPRawBody:     ocsp.MalformedRequestErrorResponse,
+		},
+	}
+	OcspInternalErrorResponse = &logical.Response{
+		Data: map[string]interface{}{
+			logical.HTTPContentType: "application/ocsp-response",
```

Comment:
nit again: why are you not using the const `ocspResponseContentType` that you defined above?